File: 소수찾기.cpp
// 에라토스테네스의 체
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n) {
    int answer = 0;
    int num = 1000000;
    int arr[1000001]={0};
    for(int i=2;i<=num;i++){
      // 2부터 시작
        arr[i]=i;
    }
    for(int i=2;i<=num;i++){
        if(arr[i]==0) continue;
        for(int j=i+i;j<=num;j+=i){
          // 2의 배수, 3의 배수 .. 지워줌
          // 2의 배수인 경우 4부터 시작
          // 2는 소수이므로
            arr[j]=0;
        }
    }
    for(int i=2;i<=n;i++){
        if(arr[i]!=0) answer++;
    }
    return answer;
}

File: 2023-08-31/소수찾기.js
function solution(n) {
  let answer = [];
  let nums = new Array(n + 1).fill(true);
  nums[0] = nums[1] = false;
  for (let i = 2; i * i <= n; i++) {
    if (nums[i]) {
      for (let j = i * i; j <= n; j += i) {
        nums[j] = false;
      }
    }
  }
  nums = nums.filter((e) => e);
  answer = nums.length;
  return answer;
}
